db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -2038,7 +2040,9 @@ Status DBImpl::FlushMemTable(ColumnFamilyData* cfd,
     WaitForPendingWrites();
 
     if (flush_reason != FlushReason::kErrorRecoveryRetryFlush &&
-        (!cfd->mem()->IsEmpty() || !cached_recoverable_state_empty_.load())) {
+        (!cfd->mem()->IsEmpty() || !cached_recoverable_state_empty_.load()) &&
+        (cfd->mem()->ApproximateMemoryUsageFast() >
```

Comment:
It should check whether the flush is triggered by `WriteBufferManager`. If a flush is triggered by write threads like split, it must flush as long as the memtable is not empty.

Reply:
`min_size_to_flush` is defaulted 0, so it will be flushed as long as the user doesn't set it otherwise.

Reply:
But `ApproximateMemoryUsageFast` is not accurate. It may still return if it's somehow 0 but there are still data in the memtable.

Reply:
In practice an empty memtable has a memory charge of ~4KB. But I'll just make `min_size_to_flush=0` a special case for peace of mind.